Multichain-proposal.md

```diff
@@ -0,0 +1,112 @@
+- SMPC nodes build and sign a Cardano transaction that **mints** k *wrapped tokens E* and sends them to the user wallet.
+
+### Sending back wrapped ETH tokens from Cardano to Ethereum
+
+The user sends their wrapped tokens to the SMPC wallet. The corresponding ETH tokens are transferred to the user wallet on the Ethereum side of the bridge. The wrapped tokens will be burnt when there is incoming transaction from Ethereum. 
+
+Flow description:
+
+- User sends **k** *wrapped E* to the SMPC wallet.
+- SMPC nodes send **k** *tokens E* to the user wallet on the ETH side.
+- k *wrapped tokens E will be burnt when there is incoming transaction.*
+
+## Bringing Cardano tokens into Ethereum
+
+### **Sending Cardano tokens to Ethereum**
+
+The user sends their cardano native tokens to the SMPC wallet, and then the SMPC nodes send the corresponding wrapped ETH tokens on the Ethereum side. The cardano native tokens keep locked until another operation needs to receive the tokens back from Ethereum. 
+
+Flow description:
+
+- User sends **k** *tokens C* to the SMPC wallet.
+- SMPC nodes send **k** *wrapped tokens C* to the user wallet on the ETH side.
+
+### Sending back wrapped Cardano tokens from Ethereum to Cardano
+
+The user sends their wrapped tokens using the ETH side of the bridge. In the Cardano side, the SMPC wallet must have locked at least the number of tokens transferred on the ETH side. This is an invariant that must hold: *every wrapped version of a cardano token in the ETH side must have the corresponding cardano token locked in the SMPC wallet.*
+Then the corresponding amount of cardano tokens is sent to the user wallet. 
+
+Flow description:
+
+- User burn **k** *wrapped C* in the ETH side.
+- SMPC nodes build and sign a cardano transaction that sends k *tokens C* to the user wallet.
+
+# Other Specification
+
+1, Decimal
+
+Because the value of token in Cardano is limited to int 64, we need to consider the decimal convert in the 3 categories. 
+
+a, If the total supply can be expressed by int 64 with the default decimal. The wrapped tokens on Cardano will keep the default decimal as native token.
+b, If the total supply can not be expressed by int 64 with the default decimal, the decimal on Cardano side will be set as 6. The Multichain protocol will convert the decimal.
+c, If the int 64 with 6 decimal still can’t express the total supply of the token, the decimal on Cardano side will be set as 0. The Multichain protocol will convert the decimal. This need to be agreed by the token project team. 
+
+2, Rate
+
+The Multichain protocol does not send two users in one transaction. All the transactions will be processed one by one. So there is only one transaction (from other chains to Cardano) in one block.
+
+3, Minimum ADA for every UTXO
+
+In Cardano, every UTXO must contain a minimal amount of ADA. After evaluation, extra fee equivalent to 2 ADA will be added to the base fee for every transaction from other chains to Cardano. 
```

Comment:
So you need some mechanism for updating the fees on Ethereum side depending on the ADA/ETH rate. How would you manage that?

Reply:
In this stage, the fee is adjusted periodically offchain. In the future version, this should be managed on chain with oracle.

Reply:
Let's add that to the document.